### ISSUE TITLE
Deprecation Warning on "._headers"

### ISSUE THREAD

**Issue by eriizu** [VilledeMontreal/express-idempotency#21]
Hello,

The package seems to work as intended, but produces the following deprecation warning:

```
(node:12797) [DEP0066] DeprecationWarning: OutgoingMessage.prototype._headers is deprecated
```

For reference, node provides the [following explanation](https://nodejs.org/api/deprecations.html#deprecations_dep0066_outgoingmessage_prototype_headers_outgoingmessage_prototype_headernames) on the matter.

I attempted to fix it myself using node's recommended public method, but can't get the project to compile as it produces the following errors.

```
> express-idempotency@1.0.4 compile
> ./node_modules/.bin/tsc --build tsconfig.dist.json

node_modules/@types/express-serve-static-core/index.d.ts:589:18 - error TS2430: Interface 'Response<ResBody, Locals, StatusCode>' incorrectly extends interface 'ServerResponse'.
  Property 'req' is optional in type 'Response<ResBody, Locals, StatusCode>' but required in type 'ServerResponse'.

589 export interface Response<
                     ~~~~~~~~

node_modules/@types/express/ts4.0/index.d.ts:43:55 - error TS2344: Type 'Response<any, Record<string, any>>' does not satisfy the constraint 'ServerResponse'.
  Property 'req' is optional in type 'Response<any, Record<string, any>>' but required in type 'ServerResponse'.

43     var static: serveStatic.RequestHandlerConstructor<Response>;
                                                         ~~~~~~~~


Found 2 errors.
```

To get the following messages, I cloned the repo, ran `npm install` and then `npm run compile`. I'll will probably look further at the cause of those errors at some point later if you don't have the time to do so, but I expect your knowledge of the project might speed up resolving this issue.

**Comment by joscelynjean** [comment]
Hello @eriizu ,

Thank you for your feedback.

Which version of Node are you using? I will try to reproduce the error on my side.

**Comment by eriizu** [comment]
Yes of course @joscelynjean

I've tried on node `v14.17.1` and `v16.3.0`.

**Comment by joscelynjean** [comment]
I have been able to reproduce the deprecation warning and the errors.

As you mention, I think it will be required to migrate from the `._headers` to the recommended public method `.getHeaders()`. I just can't remember why I used the private method. I will put it in my TODO list but feel free the submit a pull request for this.

My best bet related to the compilation errors would be it a compatibility problem between earlier version of node and the typing.  I need to go further with the investigation.
